src/state/reducers/workspace/WorkspaceReducer.js

```diff
@@ -33,10 +33,19 @@ const workspacesListInitialState = {
 };
 
 const workspacesListReducer = createReducer(workspacesListInitialState, (builder) => {
-  builder.addCase(WORKSPACE_ACTIONS_KEY.SET_ALL_WORKSPACES, (state, action) => {
-    state.data = action.list;
-    state.status = action.status;
-  });
+  builder
+    .addCase(WORKSPACE_ACTIONS_KEY.LINK_TO_DATASET, (state, action) => {
+      state.data.forEach((workspace) => {
```

Comment:
Is it not more simple to use `find` function ?

`workspaceToLink = state.data.find((workspace) => workspace.id === action.workspaceId;`
`const linkedDatasets = workspaceToLink.linkedDatasetIdList;`

Reply:
Good idea, I'll make the change :+1: